kingpin/actors/aws/s3.py

```diff
@@ -162,10 +183,23 @@ def _create_bucket(self):
         returns:
             <A boto.s3.Bucket object>
         """
+        # If we're running in DRY mode, then we create a fake bucket object
+        # that will be passed back. This mock object lets us simplify the rest
+        # of our code because we can mock out the results of creating a fresh
+        # empty bucket with no policies, versions, etc.
         if self._dry:
             self.log.warning('Would have created s3://%s' %
                              self.option('name'))
-            raise gen.Return()
+
+            # Generate a fake bucket and return it
+            mock_bucket = mock.MagicMock(name=self.option('name'))
```

Comment:
We don't want to mock the entire boto. We actually check if certain things exist, or if permissions are sufficient. Our "dry" run doesn't mean "offline test of syntax and compilation". It is more of a "sanity check everything that's possible without making the actual intended change"